### DISCUSSION TITLE
SDK updates for breaking change on query id

### DISCUSSION THREAD

**Discussion by wsjk** [looker-open-source/sdk-codegen#1514] [Q&A] [answered]
According to[ release notes for Looker 24.18](https://cloud.google.com/looker/docs/release-notes#October_09_2024) we need to use slug for `query_id` instead of numeric `id`. The release is coming Oct 14, 2024

I don't see any updates in Python SDK to handle this.

Right now we have code that gets query strings like this:

`sdk.run_query(query_id=query.id, result_format="sql")`

If I'm reading release notes correctly, this cmd will no longer work and I need to change to use `query.slug` but `run_query()` method doesn't accept string for `query_id` arg in latest SDK version

**Comment by aschott-looker** [comment, marked as answer]
The API spec is not changing as part of this change (i.e. `query_id` was and will continue to be an opaque string). The value of that string is changing. In most cases, it should not be necessary to update your code.

> If I'm reading release notes correctly, this cmd will no longer work and I need to change to use query.slug

As part of this change, `query.id` itself will return the `slug` value so this command _will_ continue to work. It should only be necessary to change call sites if the `query_id` had been hard-coded to a numeric value.

> run_query() method doesn't accept string for query_id arg in latest SDK version

It does accept a string: https://github.com/looker-open-source/sdk-codegen/blob/main/python/looker_sdk/sdk/api40/methods.py#L9153

**Reply by wsjk** [reply]
Will I at least need to use a specific version of SDK for this?

**Reply by aschott-looker** [reply]
As long as you are using a version of the SDK where `query_id` is a string, which has been the case for multiple years at this point, you should be good!

**Reply by wsjk** [reply]
cool yea we're using v22.20.0 atm

**Reply by aschott-looker** [reply]
FYI an issue was encountered while rolling this change out in `24.18`, so this was delayed. Another attempt to enable this will go out in `24.20` which should be outlined in the release notes.
